### PR TITLE
fix: reuse audio context, no audio when using krisp in some cases

### PR DESCRIPTION
**Pull request by KaustubhKumar05**
# Description

https://app.devrev.ai/100ms/works/ISS-30684

Use existing audio context in plugins manager

---

### Pre-launch Checklist

- [x] I updated/added relevant documentation.
- [x] The [Documentation] is updated accordingly, or this PR doesn't require it.
- [x] All existing and new tests are passing.
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making, or this PR is test-exempt.

### Merging:
- Squash merge to dev
- Merge commit to publish-alpha and main

[Documentation]: https://www.100ms.live/docs

### CODE REVIEW

File: packages/hms-video-store/src/utils/media.ts [hunk 1]
```diff
@@ -2,6 +2,11 @@ import HMSLogger from './logger';
 import { BuildGetMediaError } from '../error/utils';
 import { HMSTrackExceptionTrackType } from '../media/tracks/HMSTrackExceptionTrackType';
 
+//Handling sample rate error in case of firefox
+const checkBrowserSupport = () => {
+  return navigator.userAgent.indexOf('Firefox') !== -1;
```
Comment (raviteja83) [inline packages/hms-video-store/src/utils/media.ts:7]: should we ua parser here to identify this?

File: packages/hms-video-store/src/utils/media.ts [hunk 2]
```diff
@@ -48,11 +53,24 @@ export interface HMSAudioContext {
...
+  resumeContext(): Promise<void>;
+} = {
   audioContext: null,
-  getAudioContext() {
+  getAudioContext(options?: AudioContextOptions) {
     if (!this.audioContext) {
-      this.audioContext = new AudioContext();
+      if (checkBrowserSupport()) {
+        /**
+        Not setting default sample rate for firefox since connecting
```
Comment (raviteja83) [inline packages/hms-video-store/src/utils/media.ts:66]: add proper jsdoc like comment.

File: packages/hms-video-store/src/audio-sink-manager/AudioSinkManager.ts [hunk 3]
```diff
@@ -72,6 +73,7 @@ export class AudioSinkManager {
    */
   async unblockAutoplay() {
     if (this.autoPausedTracks.size > 0) {
+      HMSAudioContextHandler.resumeContext();
```
Comment (raviteja83) [inline packages/hms-video-store/src/audio-sink-manager/AudioSinkManager.ts:76]: await , below one also we should await i think

Comment (KaustubhKumar05) [reply packages/hms-video-store/src/audio-sink-manager/AudioSinkManager.ts]: Updated both

File: packages/hms-video-store/src/utils/media.ts [hunk 4]
```diff
@@ -48,11 +49,24 @@ export interface HMSAudioContext {
...
-      this.audioContext = new AudioContext();
+      if (isFirefox) {
+        /**
+         * Not setting default sample rate for firefox since connecting
+         * audio nodes from context with different sample rate is not
+         * supported in firefox
+         */
+        this.audioContext = new AudioContext();
+      } else {
+        this.audioContext = new AudioContext(options);
```
Comment (raviteja83) [inline packages/hms-video-store/src/utils/media.ts:68]: it only has three optiosn, we can check if existing sample rate is not same as the new one and create a new audio context. can be added if needed, test and let me know if this works

**Review by raviteja83 — APPROVED** [review]

**Pull request merged**